tests/client-dropped-before-command.rs

```diff
@@ -0,0 +1,37 @@
+use std::env;
+use std::process::{Command, Output};
+
+use jobserver::Client;
+
+fn main() {
+    if env::var("I_AM_THE_CLIENT").is_ok() {
+        client();
+    } else {
+        server();
+    }
+}
+
+fn server() {
+    let me = env::current_exe().unwrap();
+    let client = Client::new(1).unwrap();
+
+    let mut cmd = Command::new(me);
+    cmd.env("I_AM_THE_CLIENT", "1");
+    client.configure(&mut cmd);
```

Comment:
To test the original bug the `client` needs to be dropped after this, otherwise I think this test already passes on `mian`